### PR TITLE
solution react movies list filter

### PR DESCRIPTION
**Pull request by krolich500grn**
[DEMO LINK](https://krolich500grn.github.io/react_movies-list-filter/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+  return query.toLowerCase();
+};
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState<Movie[]>([]);
+
+  useEffect(() => {
+    setVisibleMovies(moviesFromServer);
+  }, []);
```
Comment (jstmpelowycz) [inline src/App.tsx:18]: Would you say you actually need to use `useEffect` here?

File: src/App.tsx [hunk 2]
```diff
@@ -1,9 +1,32 @@
...
+import { Movie } from './types/Movie';
+
 import moviesFromServer from './api/movies.json';
+import './App.scss';
+
+const singleCaseQuery = (query: string) => {
+  return query.toLowerCase();
+};
 
 export const App: React.FC = () => {
```
Comment (jstmpelowycz) [inline src/App.tsx:12]: I guess, you still could have applied `memo` here as well

**Review by jstmpelowycz — APPROVED** [review]
Alright!
P.S. if you still strive to optimize these components, consider applying `useCallback`, `useMemo` and so on (as we discussed)

File: src/App.tsx (hunk 1 above)
Comment (jstmpelowycz) [inline src/App.tsx:18]: You could just put the array into using `useState`

**Review by jstmpelowycz — APPROVED** [review]
Alright!
P.S. if you still strive to optimize these components, consider applying useCallback, useMemo and so on (as we discussed)

File: src/App.tsx (hunk 1 above)
Comment (krolich500grn) [reply src/App.tsx:18]: I know, but I think will be better to upload less information at the start for a faster page render

File: src/App.tsx (hunk 2 above)
Comment (krolich500grn) [reply src/App.tsx:12]: I will try to do this 👌

File: src/App.tsx [hunk 3]
```diff
@@ -1,9 +1,32 @@
...
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState<Movie[]>([]);
+
+  useEffect(() => {
+    setVisibleMovies(moviesFromServer);
```
Comment (Ihor-Karpyn) [inline src/App.tsx:17]: use moviesFromServer only here

File: src/App.tsx [hunk 4]
```diff
@@ -1,9 +1,32 @@
...
+  }, []);
+
+  const filterList = (value: string) => {
+    setVisibleMovies(moviesFromServer.filter((movie) => {
```
Comment (Ihor-Karpyn) [inline src/App.tsx:21]: Don't set in state filtered movies in state. Create function to filter. Write results in variables and then use it. Also use `useMemo` for this vara

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**